### PR TITLE
[MAINT] Replace `pytest.warns(DeprecationWarning...` with `pytest.deprecated_call()`

### LINKED ISSUE #3625: Use  `pytest.deprecated_call` to test deprecated functions
Replace the use of `with pytest.warns(DeprecationWarning)` with `with pytest.deprecated_call()`

See https://docs.pytest.org/en/latest/how-to/capture-warnings.html#ensuring-code-triggers-a-deprecation-warning

> [!TIP]
> # **💻 🍁  For the [hacktoberfest](https://hacktoberfest.com/) contributors 🍁 💻**
>
> Make sure to:
> - read our [contributing guidelines](https://nilearn.github.io/stable/development.html#contribution-guidelines)
>
> If this is your first time contributing to Nilearn, add your details in the [CITATION.cff file](https://github.com/nilearn/nilearn/blob/dcd92aa357ab3a751161aeea2edfa3a775ae644f/CITATION.cff#L42).
>
> For your pull request to be valid, you should to the following:

- [ ] remove the `check_deprecation` function in [nilearn/nilearn/_utils/testing.py](https://github.com/nilearn/nilearn/blob/1471ba887c1109d8ee8986179e0a229823a25eb6/nilearn/_utils/testing.py#L51)
- [ ] Replace the use of `with pytest.warns(DeprecationWarning)` with `with pytest.deprecated_call()` in the following

```bash
$ grep -nr "with pytest.warns(DeprecationWarning" nilearn/**/*.py

nilearn/connectome/tests/test_connectivity_matrices.py:941:    with pytest.warns(DeprecationWarning, match=match):
nilearn/datasets/tests/test_func.py:860:        with pytest.warns(DeprecationWarning):
nilearn/glm/tests/test_contrasts.py:67:    with pytest.warns(DeprecationWarning, match="0.13.0"):
nilearn/glm/tests/test_contrasts.py:262:    with pytest.warns(DeprecationWarning, match="0.13.0"):
nilearn/glm/tests/test_contrasts.py:265:    with pytest.warns(DeprecationWarning, match="0.13.0"):
nilearn/glm/tests/test_hemodynamic_models.py:60:    with pytest.warns(DeprecationWarning, match='"tr" will be removed in'):
nilearn/maskers/tests/test_nifti_labels_masker.py:220:    with pytest.warns(DeprecationWarning, match="Starting in version 0.12"):
nilearn/maskers/tests/test_nifti_maps_masker.py:186:    with pytest.warns(DeprecationWarning, match="Starting in version 0.12"):
nilearn/maskers/tests/test_nifti_masker.py:568:    with pytest.warns(DeprecationWarning, match="Starting in version 0.12"):
nilearn/maskers/tests/test_nifti_spheres_masker.py:357:    with pytest.warns(DeprecationWarning, match="Starting in version 0.12"):
nilearn/mass_univariate/tests/test_permuted_least_squares.py:752:    with pytest.warns(DeprecationWarning):
nilearn/mass_univariate/tests/test_permuted_least_squares.py:885:    with pytest.warns(DeprecationWarning):
nilearn/plotting/tests/test_html_document.py:50:    with pytest.warns(DeprecationWarning, match="temp_file_lifetime"):
nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py:112:    with pytest.warns(DeprecationWarning, match=match):
nilearn/_utils/testing.py:56:        with pytest.warns(DeprecationWarning, match=match):
```

### PR DESCRIPTION
**Pull request by victoris93**
Closes #3625

* Removed `check_deprecation` from `nilearn/nilearn/_utils/testing.py`
* Replaced `pytest.warns(DeprecationWarning...` with `pytest.deprecated_call()` in all test scripts where the former was used.

### CODE REVIEW

**Comment by victoris93** [comment]
Will update the log if the changes are ok for you

**Review by bthirion — COMMENTED** [review]
LGTM overall.
It seems that we lose something with the absence of matching pattern, but maybe that's minor ? What do people think ?

**Comment by Remi-Gau** [comment]
> we lose something with the absence of matching pattern, but maybe that's minor ? What do people think ?

oops yeah indeed: @victoris93 the `pytest.deprecated_call()` does accept an optional match argument, do you think you could keep the previous match we had?

File: nilearn/glm/tests/test_contrasts.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def test_deprecation_contrast_type(rng, set_up_glm):
     labels, results, q = set_up_glm(rng, "ar1")
     con_val = np.eye(q)[0]
 
-    with pytest.warns(DeprecationWarning, match="0.13.0"):
+    with pytest.deprecated_call():
```
Comment (Remi-Gau) [inline nilearn/glm/tests/test_contrasts.py:68]:
```suggestion
    with pytest.deprecated_call(match="0.13.0"):
```

for example

**Pull request closed without merging**